spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,253 @@
+					KafkaOperations.class,
+					ConsumerFactory.class,
+					LoggingProducerListener.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					Message.class,
+					ImplicitLinkedHashCollection.Element.class,
+					KafkaListener.class,
+					MessageMapping.class,
+					KafkaListeners.class,
+					KafkaListenerAnnotationBeanPostProcessor.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					KafkaBootstrapConfiguration.class,
+					CreatableTopic.class,
+					KafkaListenerEndpointRegistry.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS)));
+
+		Stream.of(
+					AppInfo.class,
+					// standard assignors
+					CooperativeStickyAssignor.class,
+					RangeAssignor.class,
+					RoundRobinAssignor.class,
+					StickyAssignor.class,
+					// standard partitioners
+					DefaultPartitioner.class,
+					RoundRobinPartitioner.class,
+					UniformStickyPartitioner.class,
+					// standard serialization
+					ByteArrayDeserializer.class,
+					ByteArraySerializer.class,
+					ByteBufferDeserializer.class,
+					ByteBufferSerializer.class,
+					BytesDeserializer.class,
+					BytesSerializer.class,
+					DoubleSerializer.class,
+					DoubleDeserializer.class,
+					FloatSerializer.class,
+					FloatDeserializer.class,
+					IntegerSerializer.class,
+					IntegerDeserializer.class,
+					ListDeserializer.class,
+					ListSerializer.class,
+					LongSerializer.class,
+					LongDeserializer.class,
+					StringDeserializer.class,
+					StringSerializer.class,
+					// Spring serialization
+					DelegatingByTopicDeserializer.class,
+					DelegatingByTypeSerializer.class,
+					DelegatingDeserializer.class,
+					ErrorHandlingDeserializer.class,
+					DelegatingSerializer.class,
+					JsonDeserializer.class,
+					JsonSerializer.class,
+					ParseStringDeserializer.class,
+					StringOrBytesSerializer.class,
+					ToStringSerializer.class)
+				.forEach(type -> reflectionHints.registerType(type, builder ->
+						builder.withMembers(MemberCategory.INVOKE_PUBLIC_CONSTRUCTORS)));
+
+		reflectionHints.registerType(TypeReference.of("java.util.zip.CRC32C"), builder ->
+				builder.withMembers(MemberCategory.INVOKE_PUBLIC_CONSTRUCTORS));
+
+		hints.proxies().registerJdkProxy(Consumer.class, SpringProxy.class, Advised.class, DecoratingProxy.class);
+		hints.proxies().registerJdkProxy(Producer.class, SpringProxy.class, Advised.class, DecoratingProxy.class);
+
+		if (ClassUtils.isPresent("org.apache.kafka.streams.StreamsBuilder", null)) {
```

Comment:
I guess it is preferable to use a dedicated for this purposes API on the hints: `registerTypeIfPresent()`

Reply:
The problem with that is I have to use FQ class names instead of the classes themselves.

Reply:
Are you going to fix this via `onReachableType(TypeReference.of("org.apache.kafka.streams.StreamsBuilder")` ?

Reply:
>That is correct: there is a variant for fqcn and classLoader

Right converting all those classes to FQCN is a pain.

Where is `onReachableType`?

Reply:
See how I used it in my SI PR: https://github.com/spring-projects/spring-integration/pull/3832/files#diff-b471878878aa8838786d4a93b19f8014d497db6ace582572f834da814720e01bR97